### PR TITLE
This form is being donated by Dr. Andy Gauler PhD, MS, LMHC

### PR DESCRIPTION
**Pull request by juggernautsei**
Fixes #

#### Short description of what this resolves:
They don't use the SOAP. They use the DAP form for notes. We have been carrying this form since 4.2.2. He would like it included in the codebase for the future.

### CODE REVIEW

File: contrib/forms/dap/C_FormDAP.class.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (bradymiller) [inline contrib/forms/dap/C_FormDAP.class.php:1]: need the standard header (license, author, copyright) in all these files except for table.sql

File: contrib/forms/dap/new.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+include_once("../../globals.php");
+include_once("$srcdir/api.inc");
```
Comment (bradymiller) [inline contrib/forms/dap/new.php:4]: use require_once for both lines above

File: contrib/forms/dap/report.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+//------------Forms generated from formsWiz
+
+include_once(dirname(__FILE__) . '/../../globals.php');
+include_once($GLOBALS["srcdir"] . "/api.inc");
```
Comment (bradymiller) [inline contrib/forms/dap/report.php:6]: use require_once for both lines above

File: contrib/forms/dap/report.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        print "<table><tr>";
+        foreach ($data as $key => $value) {
+            if ($key == "id" || $key == "pid" || $key == "user" || $key == "groupname" || $key == "authorized" || $key == "activity" || $key == "date" || $value == "" || $value == "0000-00-00 00:00:00") {
+                continue;
+            }
+            if ($value == "on") {
+                $value = "yes";
+            }
+            $key = ucwords(str_replace("_", " ", $key));
+            print "<td><span class=bold>" . xl($key) . ": </span><span class=text>" . text($value) . "</span></td>";
```
Comment (bradymiller) [inline contrib/forms/dap/report.php:23]: change xl() to xlt()

File: contrib/forms/dap/save.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+include_once("../../globals.php");
+include_once("$srcdir/api.inc");
```
Comment (bradymiller) [inline contrib/forms/dap/save.php:4]: use require_once for both lines above

File: contrib/forms/dap/table.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+`id` bigint(20) NOT NULL auto_increment,
+`date` datetime default NULL,
+`pid` bigint(20) default 0,
+`user` varchar(255) default NULL,
+`groupname` varchar(255) default NULL,
+`authorized` tinyint(4) default 0,
+`activity` tinyint(4) default 0,
+`data` text default NULL,
+`assessment` text default NULL,
+`plan` text default NULL,
```
Comment (bradymiller) [inline contrib/forms/dap/table.sql:11]: remove the default NULL from above text fields:
```
`data` text,
`assessment` text,
`plan` text,
```

File: contrib/forms/dap/view.php [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+include_once("../../globals.php");
+include_once("$srcdir/api.inc");
```
Comment (bradymiller) [inline contrib/forms/dap/view.php:4]: use require_once for both lines above

File: contrib/forms/dap/save.php [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
```
Comment (bradymiller) [inline contrib/forms/dap/save.php:5]:
```
use OpenEMR\Common\Csrf\CsrfUtils;

if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
    CsrfUtils::csrfNotVerified();
}
```

File: contrib/forms/dap/templates/general_new.html [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+<head>
+    <title>{xlt t='DAP'}</title>
+    {headerTemplate}
+</head>
+<body>
+<div class="container mt-3">
+    <div class="row">
+        <div class="col-12">
+            <H2>{xlt t='DAP'}</H2>
+                <form name="dap" method="post" action="{$FORM_ACTION}/interface/forms/dap/save.php">
```
Comment (bradymiller) [inline contrib/forms/dap/templates/general_new.html:11]:
```
<input type="hidden" name="csrf_token_form" value="{$CSRF_TOKEN_FORM|attr}">
```

**Comment by bradymiller** [comment]
review complete :)

**Comment by bradymiller** [comment]
hi @juggernautsei , Code looks good. Just let me know when its ready and then will bring this in. thanks! -brady

**Comment by juggernautsei** [comment]
@bradymiller at this time I have nothing I want to add unless you can think of anything. You can go ahead and bring it in.

**Comment by bradymiller** [comment]
hi @juggernautsei , bringin in. thanks for the contribution! -brady

**Pull request merged**